policy-engine/src/main.rs

```diff
@@ -72,33 +75,59 @@ async fn main() -> Result<(), Error> {
     let registry: &'static Registry = Box::leak(Box::new(metrics::new_registry(Some(
         METRICS_PREFIX.to_string(),
     ))?));
-    graph::register_metrics(registry)?;
     registry.register(Box::new(BUILD_INFO.clone()))?;
+
+    // Main service.
+    let plugins = settings.validate_and_build_plugins(Some(registry))?;
+
+    // Shared state.
+    let state = {
+        let mandatory_params = settings.mandatory_client_parameters.clone();
+        let path_prefix = settings.path_prefix.clone();
+        let plugins = Box::leak(Box::new(plugins));
+        let live = Arc::new(RwLock::new(false));
+        let ready = Arc::new(RwLock::new(false));
+
+        AppState::new(
+            mandatory_params,
+            path_prefix,
+            plugins,
+            live,
+            ready,
+            registry,
+        )
+    };
+
+    graph::register_metrics(state.registry())?;
+    let metric_state = state.clone();
     let metrics_server = HttpServer::new(move || {
         App::new()
             .wrap(middleware::Compress::default())
-            .app_data(actix_web::web::Data::new(RegistryWrapper(registry)))
+            .app_data(actix_web::web::Data::new(metric_state.clone()))
             .service(
                 actix_web::web::resource("/metrics")
-                    .route(actix_web::web::get().to(metrics::serve::<RegistryWrapper>)),
+                    .route(actix_web::web::get().to(metrics::serve::<AppState>)),
+            )
+            .service(
+                actix_web::web::resource("/liveness")
+                    .route(actix_web::web::get().to(status::serve_liveness)),
+            )
+            .service(
+                actix_web::web::resource("/readiness")
```

Comment:
A common convention in Kube-land (apparently inherited [from Google][1] [z-pages][2]) is to use [`/readyz` and `/livez` for these][3].  I'm agnostic about including the `z`, but I have a weakly-held opinion that we should avoid striking out on our own with the `ness` suffix ;)

[1]: https://github.com/kubernetes/kubernetes/pull/419#discussion_r14854689
[2]: https://stackoverflow.com/questions/43380939/where-does-the-convention-of-using-healthz-for-application-health-checks-come-f
[3]: https://kubernetes.io/docs/reference/using-api/health-checks/#api-endpoints-for-health